### PR TITLE
Add local file path support to SitemapLoader

### LINKED ISSUE #877: Sitemap loader doesn't work with local files
### 🐛 Describe the bug

Currently, the sitemap loader only works with urls and not with file present on a local filesystem. We want to add support for it like all other data loaders which support both loading from a url and loading from local file system.

Issue was reported here on slack community: https://embedchain.slack.com/archives/C05RG7CJKC6/p1698393783510609

### PR DESCRIPTION
**Pull request by Prikshit7766**
## Description

This pull request adds local file path support to the SitemapLoader class, enabling the loader to fetch data from local file paths.

- Fixes #877

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (does not change functionality, e.g. code style improvements, linting)
- [ ] Documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Please delete options that are not relevant.

- [ ] Unit Test
- [ ] Test Script (please provide)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

## Maintainer Checklist

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] Made sure Checks passed

### CODE REVIEW

File: embedchain/loaders/sitemap.py [hunk 1]
```diff
@@ -21,23 +23,34 @@
...
 
-        soup = BeautifulSoup(response.text, "xml")
+        if urlparse(sitemap_source).scheme in ("http", "https"):
+            try:
+                response = requests.get(sitemap_source)
+                response.raise_for_status()
+                soup = BeautifulSoup(response.text, "xml")
+            except requests.RequestException as e:
+                logging.error(f"Error fetching sitemap from URL: {e}")
+                return {"doc_id": "", "data": []}
```
Comment (sidmohanty11) [inline embedchain/loaders/sitemap.py:42]: let's just return here, we don't need to pass any result
```suggestion
                return
```

**Review by sidmohanty11 — CHANGES_REQUESTED** [review]
Hey @Prikshit7766, thanks a lot! LGTM, just added a minor comment

**Comment by Prikshit7766** [comment]
Thanks for the review! I've made the changes. Let me know what you think.

**Review by sidmohanty11 — APPROVED** [review]
LGTM! Thanks @Prikshit7766 !

**Pull request merged**